### ISSUE TITLE
install fails when optional dependency does not exist in registry (cannot install `@laihoe/demoparser2`)

### ISSUE THREAD

**Issue by osztenkurden** [oven-sh/bun#8639]
### What version of Bun is running?

1.1.0-canary.130+3a6318b67

### What platform is your computer?

Microsoft Windows NT 10.0.19044.0 x64

### What steps can reproduce the bug?

```bun install @laihoe/demoparser2@0.17.2```

### What is the expected behavior?

Package installs successfully

### What do you see instead?

```F:\Repositories\buntest>bun install @laihoe/demoparser2@0.17.2
[0.00ms] ".env"
bun add v1.1.0-canary.130 (3a6318b6)
  🔍 @laihoe/demoparser2-freebsd-x64 [1/5]
error: package "@laihoe/demoparser2-freebsd-x64" not found registry.npmjs.org/@laihoe%2fdemoparser2-freebsd-x64 404
  🔍 @laihoe/demoparser2-freebsd-x64 [1/5]
error: package "@laihoe/demoparser2-darwin-universal" not found registry.npmjs.org/@laihoe%2fdemoparser2-darwin-universal 404
error: @laihoe/demoparser2-freebsd-x64@0.17.2 failed to resolve
error: @laihoe/demoparser2-darwin-universal@0.17.2 failed to resolve```

### Additional information

_No response_

**Comment by paperclover** [comment]
@Electroid please do not tag non-windows issues as windows. this bug affects all platforms.

`@laihoe/demoparser2` has an optional dependency on `@laihoe/demoparser2-freebsd-x64` and other packages, which do not exist. npm silently ignores these, bun insists an error. pnpm does not install the optional dependency.

**Comment by osztenkurden** [comment]
@paperdave I wanted to ask if there is any plan on bringing this behaviour in-line to npm / pnpm? While it probably is very rare occasion that this kind of situation happens, as someone who uses this data-parsing library in day-to-day job, tihs holds me back basically to adopting Bun on all of my machines.

**Comment by dylan-conway** [comment]
@osztenkurden #11977 will fix this issue. It will most likely be included in bun `v1.1.16`

**Comment by osztenkurden** [comment]
Thank you very much!
